src/vs/platform/protocol/electron-main/protocolMainService.ts

```diff
@@ -37,7 +39,7 @@ export class ProtocolMainService extends Disposable implements IProtocolMainServ
 		// - storage    : all files in global and workspace storage (https://github.com/microsoft/vscode/issues/116735)
 		this.addValidFileRoot(environmentService.appRoot);
 		this.addValidFileRoot(environmentService.extensionsPath);
-		this.addValidFileRoot(environmentService.globalStorageHome.fsPath);
+		this.addValidFileRoot(userDataProfilesService.currentProfile.globalStorageHome.fsPath);
```

Comment:
This will not work. If we allow different windows with different profiles, then each window needs to add their global storage home depending on the profile.

I would probably turn this around and add every profile storage home here, with a listener to react to changes.

Reply:
You are right. This PR is not yet handling multiple profiles. So, we can add this in the following PR where we add this support

Reply:
If you keep that in mind 👍